### DISCUSSION TITLE
有大佬能告诉我这段代码有什么问题吗？他一直输出乱码。

### DISCUSSION THREAD

**Discussion by ningyuan7** [community/community#143453] [Programming Help]
### Select Topic Area

Question

### Body

#include <stdio.h>

int main()
{
    int n, c, l, r;

    scanf("%d %d", &n, &c);

    int a[10000], s[10000], sum[10000];

    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }

    sum[0] = a[0];
    for (int i = 1; i < n; i++) {
        sum[i] = sum[i - 1] + a[i];
    }

    for (int i = 0; i < c; i++) {
        scanf("%d %d", &l, &r);
        s[i] = sum[r] - sum[l - 1];
    }

    for (int i = 0; i < c; i++) {
        printf("%d\n", s[i]);
    }

    return 0;
}

**Comment by gaomingzhao666** [comment]
你可能没有给每一个变量赋予初始值，这几个数组所有的变量都没有给赋值，有可能是编译器直接使用内存中的数据赋值了才导致乱码。

我认为你需要在 `int a[10000],  int b[10000], int c[10000]` 的后面加上以下代码：

```c
int arrLength = 10000;
for (int i = 0; i < arrLength; i++) a[i] = 0;
for (int i = 0; i < arrLength; i++) b[i] = 0;
for (int i = 0; i < arrLength; i++) c[i] = 0;
```

**Reply by ningyuan7** [reply]
好像还是不行

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @ningyuan7 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Reply by ningyuan7** [reply]
为啥没人回答我
